src/SimpleParser.php

```diff
@@ -0,0 +1,33 @@
+<?php
+declare(strict_types=1);
+
+namespace Lcobucci\JWT;
+
+use Lcobucci\JWT\Encoding\JoseEncoder;
+use Lcobucci\JWT\Token as TokenInterface;
+use Lcobucci\JWT\Token\Parser;
+use Lcobucci\JWT\Validation\Constraint;
+use Lcobucci\JWT\Validation\SignedWith;
+use Lcobucci\JWT\Validation\ValidAt;
+use Lcobucci\JWT\Validation\Validator;
+
+final class SimpleParser
+{
+    public function parseJwt(
+        string $jwt,
+        SignedWith $signedWith,
+        ValidAt $validAt,
+        Constraint ...$constraints
+    ): TokenInterface {
+        $token = (new Parser(new JoseEncoder()))->parse($jwt);
+
+        (new Validator())->assert(
```

Comment:
Here's an interesting thing to discuss: how much of "internals" do we want to expose in this high-level API? This implementation will throw for all possible scenarios... do we want to let the user know of that or should we return `null` for the "errors"?

If `null` is fine, then we need to put a try/catch around the parser and should probably use `validate()` instead of `assert()`.

Reply:
Let's step back for a moment: what are (99% of) the real scenarios this code raises an exception for?

1. The developer is messing things up during development
2. Someone wants to hack into production
3. Production server's clock is messed up

For all the above, I'd like to have the full exception to be thrown.

In regular runs, I have never dealt bad path validation, do you?

Reply:
I have no idea what you are asking for :sweat:

Reply:
The fundamental difference between `Validator#assert()` and `Validator#validate()` is that the former goes through all the constraints and aggregates the violations (and their messages), while the latter returns `false` at the very failed constraint.

What I'm trying to discuss here is if, for this high-level API, we should go through all constraints when the first one has failed. In my projects, I don't feel that need, ergo my question.

Reply:
My use cases tell me the opposite: most of (my) failures come from misconfiguration and server misalignment; in all scenarios, even for the few malicious targeted attempts I underwent, the more information I had the sooner I fixed the production.

Reply:
Let's leave it like this, then, my use-cases are generally not the default for people 😛 

This does highlight a small oversight on the validation API... we only have the error msg from the constraint but no reference to which constraint caused the violation. I'll have a look at this in another PR.